### PR TITLE
rbd: support namespaces for image migration

### PR DESCRIPTION
**Pull request by dillaman**
Fixes: http://tracker.ceph.com/issues/26951
Signed-off-by: Jason Dillaman <dillaman@redhat.com>

- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

**Comment by dillaman** [comment]
@trociny Good catch -- added

**Comment by dillaman** [comment]
@trociny fixed

File: src/include/rbd/librbd.h [hunk 1]
```diff
@@ -241,9 +241,11 @@ typedef enum {
 
 typedef struct {
   int64_t source_pool_id;
+  char *source_pool_namespace;
   char *source_image_name;
   char *source_image_id;
   int64_t dest_pool_id;
+  char *dest_pool_namespace;
```
Comment (trociny) [inline src/include/rbd/librbd.h]: @dillaman need to update `ctypedef struct rbd_image_migration_status_t` in rbd.pyx:
```
zhuzha:~/ceph/ceph.ci/build% RBD_FEATURES=127 PYTHONPATH=../src/test/pybind:../src/pybind:./lib/cython_modules/lib.2 LD_LIBRARY_PATH=./lib nosetests -v test_rbd:TestMigration.test_migration
test_rbd.TestMigration.test_migration ... ERROR

======================================================================
ERROR: test_rbd.TestMigration.test_migration
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/ssd/ceph/ceph.ci/src/test/pybind/test_rbd.py", line 2041, in test_migration
    status = RBD().migration_status(ioctx, image_name)
  File "rbd.pyx", line 1413, in rbd.RBD.migration_status (/home/mgolub/ceph/ceph.ci/build/src/pybind/rbd/rbd.c:11588)
OSError: [errno 34] error getting migration status
```

Comment (dillaman) [reply src/include/rbd/librbd.h]: Fixed -- and just learned that Cython doesn't validate the structs.

**Pull request merged**